src/mapping.ts

```diff
@@ -0,0 +1,275 @@
+    return tier;
+  } else if (_tier == BigInt.fromI32(2)) {
+    tier = "TWO";
+  } else if (_tier == BigInt.fromI32(3)) {
+    tier = "THREE";
+  } else if (_tier == BigInt.fromI32(4)) {
+    tier = "FOUR";
+  }
+
+  return tier;
+}
+
+// /**
+//  * @notice object type where properties represent value-based-decoded nft metadata
+//  * TODO: not sure but I am getting a compile error here. Asked DK. I need to just look into this. This will be done in next PR as I clean up the code.
+//  */
+// type Values = {
+//   massSize: BigInt;
+//   tier: string;
+//   color: string;
+// };
+
+// /**
+//  * @notice carries out decoding of value for a particular mass NFT
+//  * @returns value : Values
+//  * @dev TODO: see Object Type Values above and the error that is there. Once that is resolved then this can be used.
+//  */
+// function decodeValue(nftValue: BigInt): Values {
+//   let value: Values;
+//   let _tier: BigInt = nftValue / CLASS_MULTIPLIER;
+//   let tier = checkMergeClass(_tier);
+
+//   value = {
+//     massSize: (nftValue % CLASS_MULTIPLIER) as BigInt,
+//     tier: tier,
+//     color: checkColor(tier),
+//   };
+
+//   return value;
+// }
+
+/* ========== EVENT HANDLERS ========== */
+
+/**
+ * @notice handles Transfer events and ultimately instantiates / updates User and NFT entities
+ * @param event: Transfer
+ * @dev all NFT entity generation is handled in other functions.
+ */
+export function handleTransfer(event: Transfer): void {
+  const to = event.params.to.toHex();
+  const from = event.params.from.toHex();
+
+  let nft: NFT;
+
+  let user = User.load(to);
+  const tokenId = event.params.tokenId.toString();
+
+  // Step 0: set NG_OMNIBUS as a whitelisted address as it is done in the constructor
+  if (!user && to == NG_OMNIBUS) {
+    user = new User(to);
+    user.whitelist = true;
+  }
+
+  // Step 1: checks if this is a new user, takes care of first time NG mints too.
+  if (!user) {
+    user = new User(to);
+  } // checks if this is a mint() scenario
+
+  user.save();
+
+  // Step 2a: either create new NFT if in minting phase
+  if (from == EMPTY_ADDRESS && to == NG_OMNIBUS) {
+    nft = createNFT(event, user.id);
+  }
+
+  // Step 2b: OR updateNFT for all other Transfer scenarios incl: temporary transfers right before merging/burning, simple burning (direct to dead address, which then transfers to address(0))
+  else {
+    nft = updateNFT(tokenId, user.id);
+  }
+
+  nft.save();
+}
+
+/**
+ * @notice updates when new alphaMass identified
+ * @param event AlphaMassUpdate
+ * @dev TODO: need to handle the old alphaMass still. So if one tokenId is alpha one day, but the next a different one is, they will both have the 'isAlpha' field as 'true'
+ */
+export function handleAlphaMassUpdate(event: AlphaMassUpdate): void {
+  const alphaTokenId = event.params.tokenId.toString();
+  const nft = NFT.load(alphaTokenId) as NFT;
+
+  nft.massSize = event.params.alphaMass;
+  nft.isAlpha = true;
```

Comment:
we also need to figure out a way to remove the old alpha mass. please do this in one of the future PRs

Reply:
NOTE - You can fix this in a new PR if you want

Reply:
Yep, after going through your latest comments it hit me that PR#4 (historic data focus) chats we had before revolved around keeping track of the old Alphas.